File: ecopoints/app/src/main/java/at/htl/ecopoints/navigation/BottomNavItem.java
package at.htl.ecopoints.navigation;

import at.htl.ecopoints.R;

public enum BottomNavItem {
    HOME("Home", R.drawable.ic_home, "home_route"),
    RANKING("Ranking", R.drawable.ic_ranking, "ranking_route"),
    PROFILE("Profile", R.drawable.ic_profile, "profile_route");

    private final String title;
    private final int icon;
    private final String route;

    BottomNavItem(String title, int icon, String route) {
        this.title = title;
        this.icon = icon;
        this.route = route;
    }

    public String getTitle() {
        return title;
    }

    public int getIcon() {
        return icon;
    }

    public String getRoute() {
        return route;
    }
}

File: ecopoints/app/src/main/java/at/htl/ecopoints/navigation/BottomNavItem.kt
package at.htl.ecopoints.navigation

import at.htl.ecopoints.R

sealed class BottomNavItem(var title:String, var icon:Int, var screen_route:String){

    class Home : BottomNavItem("Home", R.drawable.ic_home, "home_route")
    class Ranking : BottomNavItem("Ranking", R.drawable.ic_ranking, "ranking_route")
    class Profile : BottomNavItem("Profile", R.drawable.ic_profile, "profile_route")
}